src/scipp/optimize/__init__.py

```diff
@@ -33,6 +33,19 @@ def func(x, *args):
     return func
 
 
+def _get_sigma(da):
+    if da.variances is None:
+        return None
+
+    sigma = stddevs(da).values
+    if not sigma.all():
+        raise ValueError(
+            'There is a 0 in the input variances. This would break the optimizer. '
```

Comment:
In the issue we had discussed raising an extra warning. Is the result totally meaningless, i.e., is raising justified?

Reply:
It seems that `method='lm'` is likely to produce nonsense results as shown in the issue. The other two methods actually raise an error because they fail to estimate residuals.
I can change it to a warning, but I see little change for the result to be meaningful.

Reply:
Makes sense, leave it as an error.